.travis.yml

```diff
@@ -2,22 +2,24 @@
 language: c++
 sudo: required
 
-os:
-  - linux
-  - osx
+services:
+  - docker
 
-dist: trusty # Ubuntu 14.04.
+before_install:
+  - docker pull aaelbashandy/travisci_ubuntu_1804_teca:0.01
+  - >
+    docker run -v $TRAVIS_BUILD_DIR:/travis_host aaelbashandy/travisci_ubuntu_1804_teca:0.01 /bin/bash -c 
+    "source /teca_deps/2.1.3/bin/teca_env.sh &&
+    export LD_LIBRARY_PATH=/teca_deps/2.1.3/lib/:/teca_deps/2.1.3/lib64/:/project/TECA_build/lib:$LD_LIBRARY_PATH &&
+    export PYTHONPATH=/teca_deps/2.1.3/lib:/teca_deps/2.1.3/lib/python2.7/site-packages/:/project/TECA_build/lib:$PYTHONPATH &&
+    svn co svn://missmarple.lbl.gov/work3/teca/TECA_data TECA_data && cd TECA_build/ &&
+    cmake -DBUILD_TESTING=ON -DTECA_DATA_ROOT=/project/TECA_data/ -DCMAKE_BUILD_TYPE=Debug
+    -DCMAKE_INSTALL_PREFIX=/teca_deps/2.1.3 -DBOOST_ROOT=/teca_deps/2.1.3/include/
+    -DNETCDF_DIR=/teca_deps/2.1.3 -DTECA_PYTHON_VERSION=2 /travis_host &&
+    make -j4 && ctest -I -V"
 
-compiler:
-  - gcc
 
-install: ./test/travis_ci/install_$TRAVIS_OS_NAME.sh
-
-env:
-  - BUILD_TYPE=Debug
-#  - BUILD_TYPE=Release
-
-script: ./test/travis_ci/ctest_$TRAVIS_OS_NAME.sh
+script: ctest -I -V
```

Comment:
I think we'll need to pass the cmake configs to ctest. These have ctest options that get the branch name into the web UI, among other features. I think you'll need to modify some parts of those scripts, checkout `test/travis_ci/ctest_linux.cmake` and `test/travis_ci/ctest_osx.cmake`